store/generated_key/v2/eigenda.go

```diff
@@ -141,12 +149,59 @@ func (e Store) BackendType() common.BackendType {
 //
 // Since v2 methods for fetching a payload are responsible for verifying the received bytes against the certificate,
 // this Verify method only needs to check the cert on chain. That is why the third parameter is ignored.
-func (e Store) Verify(ctx context.Context, certBytes []byte, _ []byte) error {
+func (e Store) Verify(ctx context.Context, certBytes []byte, _ []byte, opts common.VerifyOpts) error {
 	var eigenDACert coretypes.EigenDACert
 	err := rlp.DecodeBytes(certBytes, &eigenDACert)
 	if err != nil {
 		return fmt.Errorf("RLP decoding EigenDA v2 cert: %w", err)
 	}
+	err = e.verifyRBNRecencyCheck(eigenDACert.BatchHeader.ReferenceBlockNumber, opts)
+	if err != nil {
+		return fmt.Errorf("punctuality check failed: %w", err)
+	}
 
 	return e.verifier.VerifyCertV2(ctx, &eigenDACert)
 }
+
+// Certs in the rollup batcher-inbox that do not respect the below equation are discarded.
+//
+//	cert.RBN < cert.L1InclusionBlock <= cert.RBN + RBNRecencyWindowSize
```

Comment:
Can't remember if we'd had this discussion already-

Is `cert.RBN < cert.L1InclusionBlock` the right condition? Why shouldn't we permit `cert.RBN <= cert.L1InclusionBlock`?

Reply:
You had pointed this out actually, its impossible to have =, because RBN is used by validators to verify the dispersal request and sign on it. So has to be in the past. And all of this happens before the cert can be sent to batcher inbox